build-farm/make-adopt-build-farm.sh

```diff
@@ -28,8 +28,18 @@ JAVA_FEATURE_VERSION=$(echo "${JAVA_TO_BUILD}" | tr -d "[:alpha:]")
 
 if [ -z "${JAVA_FEATURE_VERSION}" ]
 then
-    # THIS NEEDS TO BE UPDATED WHEN HEAD UPDATES (the latest tag that jdk/jdk contains)
-    JAVA_FEATURE_VERSION=15
+  # Use Adopt API to get the JDK Head number
+  echo "This appears to be JDK Head. Querying the Adopt API to get the JDK HEAD Number (https://api.adoptopenjdk.net/v3/info/available_releases)..."
+  JAVA_FEATURE_VERSION=$(curl https://api.adoptopenjdk.net/v3/info/available_releases | jq '.most_recent_feature_version')
+    
+  # Checks the api request was successfull and the return value is a number
+  isNum='^[0-9]+$'
+  if [ -z "${JAVA_FEATURE_VERSION}" ] || ! [[ $JAVA_FEATURE_VERSION =~ $isNum ]]
```

Comment:
Is that if bracketing correct for bash?

Reply:
It has to be double bracketed as regex's don't seem to work in singular ones (tested locally on mac)

Reply:
You cold possibly get away with

`if [ "${JAVA_FEATURE_VERSION}" -gt 0 ]`

Which would fail one way or another on all the situations we're trying to trap